### PR TITLE
Add support to `--function` to rmc script (#169)

### LINKED ISSUE #169: Allow entry-point for `rmc` other than `main`
CBMC allows us to specify an entry-point other than `main`, which would be useful for analyzing a library crate, for example. However, if we try to do this, we get an error from `rmc-rustc`:

File (test.rs):
```
fn main2() {
    assert!(true);
}
```

Command: `rmc --verbose test.rs -- --function main2`

Error:
```
[RMC] stage: compile (FAIL)
[RMC] cmd: rmc-rustc -Z codegen-backend=gotoc --cfg=rmc -o test.json test.rs
error[E0601]: `main` function not found in crate `test`
 --> test.rs:1:1
  |
1 | / fn main2() {
2 | |     assert!(true);
3 | | }
  | |_^ consider adding a `main` function to `test.rs`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0601`.
```

### PR DESCRIPTION
**Pull request by celinval**
This changes the compilation to use crate type lib instead of binary when
we are running rmc on a single .rs file. This allow us to use
any public function as a verification target.

We have also changed the tests to use pub main so it is exported and it
can be used as the entry point of the proof.

Fix cargo-rmc and the current testcase to support library build.

### Description of changes: 

1- RMC users will need to ensure their main function is public in order to use them as verification harness. Note that this not affect proof harness with attribute `#[no_mangle]` since this attribute actually exports the function.
2- Users can now invoke rmc with the `--function` attribute. They can select any public function as their starting point. Note that the function still has to have the `#[no_mangle]` attribute.

### Resolved issues:

Resolves #169 

### Testing:

* How is this change tested?

* Is this a refactor change?

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by tedinski — APPROVED** [review]
I was **just** writing docs about `--function`.

I'll add the note about `#[no_mangle]` to what I wrote.

File: src/tools/dashboard/src/util.rs [hunk 1]
```diff
@@ -185,7 +185,16 @@ pub fn add_codegen_job(litani: &mut Litani, test_props: &TestProps) {
...
-        .args(["-Z", "codegen-backend=gotoc", "--cfg=rmc", "--out-dir", "build/tmp"])
+        .args([
+            "-Z",
+            "codegen-backend=gotoc",
+            "--cfg=rmc",
+            "--out-dir",
+            "build/tmp",
+            "-Z",
+            "human_readable_cgu_names",
+            "-crate-type=lib",
```
Comment (zhassan-aws) [inline src/tools/dashboard/src/util.rs:196]: Should be two dashes.

Comment (celinval) [reply src/tools/dashboard/src/util.rs:196]: Good catch... I wonder when this gets run. I should check with @adpaco-aws to make sure we have test coverage in our regression.

Comment (tedinski) [reply src/tools/dashboard/src/util.rs:196]: We have the dashboard set to only run after merges. I think there are concerns right now about prolonging CI times.

Comment (celinval) [reply src/tools/dashboard/src/util.rs]: Anyway, I fixed it and I ran the dashboard script locally. The output is a bit hard to parse, but it seemed to work.

**Review by zhassan-aws — APPROVED** [review]
